src/main/java/me/crespel/runtastic/mapper/GpxSportSessionMapper.java

```diff
@@ -46,13 +51,30 @@ public boolean supports(String format) {
 
 	@Override
 	public GpxType mapSportSession(SportSession session, String format) {

```

Comment:
This method is becoming a bit long, it might be better to split certain parts to specific methods (e.g. mapTrack, mapPhotos, mapBounds, mapMetadata ...)

Reply:
Moved part into mapMetadata(SportSession session). Further simplifications to be discussed.